### PR TITLE
update tests to be specific about the versions they are testing

### PR DESCRIPTION
**Pull request by deads2k**
When setting up tests, you want to rely on your own scheme.  This eliminates coupling to floating versions which gives unnecessary flexibility in most cases and prevents testing all the versions you need.

@liggitt  scrubs unnecessary deps.

```release-note
NONE
```

### CODE REVIEW

**Comment by deads2k** [comment]
@kubernetes/sig-api-machinery-pr-reviews 
@caesarxuchao

File: pkg/apis/core/install/install_test.go [hunk 1]
```diff
@@ -52,15 +53,15 @@ func TestCodec(t *testing.T) {
 	pod := internal.Pod{}
 	// We do want to use package registered rather than testapi here, because we
```
Comment (liggitt) [inline pkg/apis/core/install/install_test.go]: remove comment

File: pkg/api/testapi/testapi.go [hunk 2]
```diff
@@ -74,33 +74,23 @@ import (
 )
 
 var (
-	Groups        = make(map[string]TestGroup)
-	Default       TestGroup
-	Authorization TestGroup
```
[the remark this answers is not in the document]
Comment (deads2k) [reply pkg/api/testapi/testapi.go]: > did these just happen to be the unused ones, and the others will get trimmed over time?

correct

File: pkg/api/testing/conversion_test.go [hunk 3]
```diff
@@ -46,11 +47,11 @@ func BenchmarkPodConversion(b *testing.B) {
 	scheme := legacyscheme.Scheme
 	for i := 0; i < b.N; i++ {
 		pod := &items[i%width]
-		versionedObj, err := scheme.UnsafeConvertToVersion(pod, legacyscheme.Registry.GroupOrDie(api.GroupName).GroupVersions[0])
+		versionedObj, err := scheme.UnsafeConvertToVersion(pod, schema.GroupVersion{Group: "", Version: "v1"})
```
[the remark this answers is not in the document]
Comment (deads2k) [reply pkg/api/testing/conversion_test.go]: > any reason not to use core/v1.SchemeGroupVersion and core.SchemeGroupVersion, etc?

No real reason for the dependency across the stack.

**Comment by liggitt** [comment]
/lgtm

glad to see the tests getting simpler and more direct

**Pull request merged**